felix.go

```diff
@@ -778,3 +809,51 @@ func (fc *DataplaneConnector) Start() {
 	// Start background thread to read messages from dataplane driver.
 	go fc.readMessagesFromDataplane()
 }
+
+var ErrServiceNotReady = errors.New("Kubernetes service missing IP or port.")
+
+func discoverTyphaAddr(configParams *config.Config) (string, error) {
+	if configParams.TyphaAddr != "" {
+		// Explicit address; trumps other sources of config.
+		return configParams.TyphaAddr, nil
+	}
+
+	if configParams.TyphaK8sServiceName == "" {
+		// No explicit address, and no service name, not using Typha.
+		return "", nil
+	}
+
+	// If we get here, we need to look up the Typha service using the k8s API.
+	// TODO Typha: support Typha lookup without using rest.InClusterConfig().
+	k8sconf, err := rest.InClusterConfig()
+	if err != nil {
+		log.WithError(err).Error("Unable to create Kubernetes config.")
+		return "", err
+	}
+	clientset, err := kubernetes.NewForConfig(k8sconf)
+	if err != nil {
+		log.WithError(err).Error("Unable to create Kubernetes client set.")
+		return "", err
+	}
+	svcClient := clientset.CoreV1().Services(configParams.TyphaK8sNamespace)
```

Comment:
Does namespace need to be non-"" ?  If so, we should check that above with TyphaK8sServiceName.